README.md

```diff
@@ -1,3 +1,266 @@
+  These are also connected to the global SCION network.
+
+
+
+## DatagramChannel
+
+### Destinations
+In order to find a path to a destination IP, a `DatagramChannel` or `DatagramSocket` must know the 
+ISD/AS numbers of the destination.
+
+
+If the destination host has a DNS TXT entry for SCION then this be used to determine the 
+destination ISD/AS.
+Alternatively, the ISD/AS can be specified explicitly.
+
+#### DNS lookup
+```
+$ dig TXT ethz.ch
+...
+ethz.ch.		610	IN	TXT	"scion=64-2:0:9,129.132.230.98"
+...
+```
+
+```
+InetSocketAddress serverAddress = new InetSocketAddress("ethz.ch", 80);​
+channel.connect(serverAddress);
+```
+
+
+#### Explicit ISD/AS specification
+
+We can use the ISD/AS directly to request a path:
+```
+long isdAs = ScionUtil.parseIA("64-2:0:9");
+InetSocketAddress serverAddress = new InetSocketAddress("129.132.19.216", 80);​
+Path path = Scion.defaultService().getPaths(isdAs, serverAddress).get(0);
+channel.connect(path);
+```
+
+
+### Demo application - ping pong
+
+There is a simple ping pong client-server application in `src/test/demo`.
+
+It has some hardcoded ports/IP so it works only with the scionLab tiny.topo and only with the dispatcher-free
+version of scionLab: https://github.com/scionproto/scion/pull/4344
+
+The client and server connects directly to the border router (without dispatcher).
+
+The server is located in `1-ff00:0:112` (IP `[::1]:44444`). The client is located in `1-ff00:0:110`.
+
+### Options
+
+Options are defined in `ScionSocketOptions`, see javadoc for details.
+
+| Option                        | Default | Short description                                               |
+|-------------------------------|---------|-----------------------------------------------------------------|
+| `SN_API_WRITE_TO_USER_BUFFER`    | `false` | Throw exception when receiving an invalid packet          | 
+| `SN_PATH_EXPIRY_MARGIN` | `2`     | A new path is requested if `now + margin > pathExpirationDate` | 
+
+## Performance pitfalls
+
+- **Using `SocketAddress` for `send()`**. `send(buffer, socketAddress)` is a convenience function. However, when sending 
+  multiple packets to the same destination, one should use `path = send(buffer, path)` or `connect()` + `write()` in 
+  order to avoid frequent path lookups.
+
+- **Using expired path (client).** When using `send(buffer, path)` with an expired `RequestPath`, the channel will 
+  transparently look up a new path. This works but causes a path lookup for every `send()`.
+  Solution: always use the latest path returned by send, e.g. `path = send(buffer, path)`.
+
+- **Using expired path (server).** When using `send(buffer, path)` with an expired `ResponsePath`, the channel will
+  simple send it anyway.
+
+## Configuration
+
+### Bootstrapping / daemon
+In order to find paths and connect to the local AS, the application needs either a (local) 
+installation of the Scion Daemon (see here) or some other means to
+get bootstrap information.
```

Comment:
Imprecise/overloaded wording: The bootstrap info consists of the topology.json and the certs. Usually you thus need the bootstrap info plus the daemon to get running. The daemon can't do the bootstrapping on its own.